src/subdomains/core/payment-link/services/payment-link.service.ts

```diff
@@ -242,11 +242,29 @@ export class PaymentLinkService {
     if (country) updatePaymentLink.country = await this.countryService.getCountryWithSymbol(country);
     if (country === null) updatePaymentLink.country = null;
 
-    await this.paymentLinkRepo.update(paymentLink.id, updatePaymentLink);
+    await this.updatePaymentLinkInternal(paymentLink, updatePaymentLink);
 
     return this.getOrThrow(userId, linkId, externalLinkId, externalPaymentId);
   }
 
+  async updatePaymentLinkAdmin(id: number, dto: UpdatePaymentLinkInternalDto): Promise<PaymentLink> {
+    const entity = await this.paymentLinkRepo.findOneBy({ id });
+    if (!entity) throw new NotFoundException('PaymentLink not found');
+
+    return this.updatePaymentLinkInternal(entity, dto);
+  }
+
+  async updatePaymentLinkInternal(paymentLink: PaymentLink, dto: Partial<PaymentLink>): Promise<PaymentLink> {
+    if (dto.country) {
+      dto.country = await this.countryService.getCountry(dto.country.id);
+      if (!dto.country) throw new NotFoundException('Country not found');
+    }
```

Comment:
This can be moved to `updatePaymentLinkAdmin`? Is already done in `update` method